### PR TITLE
Add Show instances, printCSSLength and printFontSize

### LINKED ISSUE #22: Print and show
Some attribute types have `Show` instances (e.g., `CSSLength`), others have `printX` functions (e.g., `Align`) and others still have both (e.g., `MarkerUnit`).

I propose that every attribute type have a `Show` instance that works the "standard" way. That is, it follows the convention laid out in the `class Show` definition that:
> While not required, it is recommended that for any expression x, the string show x be executable PureScript code which evaluates to the same value as the expression x.

Also there should be a `Print` class:
```
class Print a where
  print :: a -> String
```
and every attribute type should have a `Print` instance.

Having `Print` isn't just a nice naming convention. It would allow the definition of a helper function:
```
mkAttr :: forall r i a. Print a => String -> a -> IProp r i
mkAttr name = attr (AttrName name) <<< print
```
so that most of our attribute definitions would look like:
```
x :: forall r i. Number -> IProp (x :: Number | r) i
x = mkAttr "x"
```

### PR DESCRIPTION
**Pull request by JamieBallingall**
With this PR every attribute type now has:
- An `Eq` instance
- A `Show` instance
- A `printX` function that produces what is needed by SVG

The documentation for `Show` says "While not required, it is recommended that for any expression x, the string show x be executable PureScript code which evaluates to the same value as the expression x." This is true for all types that have Show instances except `CSSLength` and `FontSize`. Fixing `CSSLength` and `FontSize` will be a (minor) breaking change.

This fixes #22 and I'll add a separate issue to track changes to `CSSLength` and `FontSize`.

### CODE REVIEW

**Comment by JordanMartinez** [comment]
I'm not sure we need this comment below to reappear multiple times.
```
{- The documentation for Show says "While not required, it is recommended that
for any expression x, the string show x be executable PureScript code which
evaluates to the same value as the expression x." The instance of Show above
predates the use of printFontSize and does not follow that recommendation. This
may be changed in a future version and `show` when generating SVG is
depreciated. Use printFontSize instead.
-}
```
Rather, I think it should be stated more generally in the `Attributes.purs` file as a module documentation. Something like
> Each attribute has two functions: `show` from the type class `Show`
> and `printX` where `X` corresponds to the type. When printing the value
> for SVG attributes, use the `printX` function. For example
> ... example code block...
> When debugging, use either `show` or `printX`.

**Comment by JordanMartinez** [comment]
Otherwise, this looks good to me!

**Comment by JordanMartinez** [comment]
Can you add the module documentation comment? Then I'll merge this PR.

**Comment by JamieBallingall** [comment]
Sorry about the passive-aggressive CHANGELOG commit. It suddenly occurred to me and I didn't see that you'd already commented.

I've put that Attributes level comment in but I've left smaller comments on CSSLength and FontSize to call out that they are different and are likely to change. Turns out similar comments on MarkerUnit and Orient are also required.

File: src/Halogen/Svg/Attributes.purs [hunk 1]
```diff
@@ -92,6 +92,12 @@ import Halogen.Svg.Attributes.Path (
   m, l, h, v, c, s, q, t, a, z
   )
 
+-- Each attribute has two functions: show from the type class Show
+-- and printX where X corresponds to the type. When printing the value
+-- for SVG attributes, use the printX function. For example:
+--   `fill = attr (H.AttrName "fill") <<< printColor`
+-- When debugging, use either show or printX.
+
```
Comment (JordanMartinez) [inline src/Halogen/Svg/Attributes.purs:100]: I meant adding a doc comment [on the Attribute.purs module](https://github.com/JordanMartinez/purescript-halogen-svg-elems/blob/master/src/Halogen/Svg/Attributes.purs#L1)

For example:
```purescript
-- | Each attribute has ....
module Halogen.Svg.Attributes where
---
```

File: src/Halogen/Svg/Attributes.purs [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
```
Comment (JordanMartinez) [inline src/Halogen/Svg/Attributes.purs:6]: Could you make these `-- | ` so that they show up as documentation the module?

**Review by JordanMartinez — CHANGES_REQUESTED** [review]

**Comment by JordanMartinez** [comment]
Thanks!

**Pull request merged**